libevmasm/AssemblyItem.cpp

```diff
@@ -352,13 +352,13 @@ std::string AssemblyItem::toAssemblyText(Assembly const& _assembly) const
 		break;
 	case PushTag:
 	{
-		size_t sub{0};
+		SubAssemblyID sub;
 		size_t tag{0};
 		std::tie(sub, tag) = splitForeignPushTag();
```

Comment:
At first I wanted to point out that you're changing the initial value from `0` to `max()`, but then I noticed it's getting overwriten anyway. This would be clearer:

```suggestion
		auto [sub, tag] = splitForeignPushTag();
```

Reply:
Same in the `tagRef` loop in `assembleLegacy()`.

Reply:
fixed